contracts/examples/esdt-transfer-with-fee/src/esdt_transfer_with_fee.rs

```diff
@@ -0,0 +1,105 @@
+#![no_std]
+
+mod fee;
+use fee::*;
+
+elrond_wasm::imports!();
+#[elrond_wasm::contract]
+pub trait EsdtTransferWithFee {
+    #[init]
+    fn init(&self) {}
+
+    #[only_owner]
+    #[endpoint(setFeeForToken)]
+    fn set_fee_for_token(&self, fee: Fee<Self::Api>, token: TokenIdentifier) {
```

Comment:
I think this should be split into two endpoints, one for percentage, one for fixed. It's pretty difficult to encode the arguments otherwise.